solidity_contracts/src/CairoPrecompiles/PragmaCaller.sol

```diff
@@ -71,4 +130,111 @@ contract PragmaCaller {
         }
         return response;
     }
+
+    function calculateMean(PragmaCalculateMeanRequest memory request)
+        public
+        view
+        returns (PragmaSummaryStatsResponse memory response)
+    {
+        // Serialize the data request into a format compatible with the expected Pragma inputs
+        uint256[] memory data = new uint256[](request.dataType == DataType.FuturesEntry ? 6 : 5);
+        data[0] = uint256(request.dataType);
+        data[1] = request.pairId;
+        if (request.dataType == DataType.FuturesEntry) {
+            data[2] = request.expirationTimestamp;
+            data[3] = request.startTimestamp;
+            data[4] = request.endTimestamp;
+            data[5] = uint256(request.aggregationMode);
+        } else {
+            data[2] = uint256(request.startTimestamp);
+            data[3] = uint256(request.endTimestamp);
+            data[4] = uint256(request.aggregationMode);
+        }
```

Comment:
`startTimestamp` and `endTimestamp` are uint64 but data is an array of uint256.
They are casted in only the else

Reply:
Ah true, removed the cast!